src/Services/Azure/Authentication/CustomChainedCredential.cs

```diff
@@ -119,4 +135,76 @@ private static DefaultAzureCredential CreateDefaultCredential(string? tenantId)
 
         return new DefaultAzureCredential(defaultCredentialOptions);
     }
+
+    private static TokenCredential? CreateVsCodeBrokerCredential(string? tenantId)
+    {
+        const string vsCodeClientId = "aebc6443-996d-45c2-90f0-388ff96faa56";
+        string userProfile = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+        string authRecordPath = Path.Combine(userProfile, ".azure", "ms-azuretools.vscode-azureresourcegroups", "authRecord.json");
+        if (!File.Exists(authRecordPath))
+        {
+            // Try .Azure if .azure is not present
+            authRecordPath = Path.Combine(userProfile, ".Azure", "ms-azuretools.vscode-azureresourcegroups", "authRecord.json");
+            if (!File.Exists(authRecordPath))
+            {
+                return null;
+            }
+        }
+
+        AuthenticationRecord? authRecord;
+        try
+        {
+            using var stream = File.OpenRead(authRecordPath);
+            authRecord = AuthenticationRecord.Deserialize(stream);
+        }
+        catch
+        {
+            // Deserialization failed
```

Comment:
I feel these cases need to be logged by azmcp for debugging

Reply:
Logging is added.